core/App/components/misc/CredentialCard.tsx

```diff
@@ -34,30 +33,69 @@ const paddingVertical = 10
 const paddingHorizontal = 10
 const transparent = 'rgba(0,0,0,0)'
 const borderRadius = 15
+const borderPadding = 8
 const { width } = Dimensions.get('window')
 
+/**
+ * A card is defined as a 4x8 (height/rows x width/columns) grid.
+ | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 |
+ | 2 |   |   |   |   |   |   |   |
+ | 3 |   |   |   |   |   |   |   |
+ | 4 |   |   |   |   |   |   |   |
+ 
+The card width is the full screen width, and the card height is half of the screen width
+
+Variation 1:
+  Header: Small Logo (1x1) + Issuer Name (1x3) + Credential Name (1x4)
+  Body: Reserved for Future Use (2x4)
+  Footer: Issued or Expired Date (1x4)
+
+  | L | Issuer    | Cred Name     |
+  |             Body              |
+  |             Body              |
+  | Issued/Expired Date           |
+
+ Variation 2:
+  Header: Large Logo (1x4) + Credential Name (1x4)
+  Body: Reserved for Future Use (2x4)
+  Footer: Issued or Expired Date (1x4)
+
+  | Logo          | Cred Name     |
+  |             Body              |
+  |             Body              |
+  | Issued/Expired Date           |
+
+
+Note: The small logo MUST be provided as 1x1 (height/width) ratio, while the large logo MUST be provided as 1x4 (height/width) ratio
+ */
+const cardHeight = width / 2 // a card height is half of the screen width
+const cardHeaderHeight = cardHeight / 4 // a card has a total of 4 rows, and the header occupy 1 row
+
 const toImageSource = (source: unknown): ImageSourcePropType => {
   if (typeof source === 'string') {
     return { uri: source as string }
   }
   return source as ImageSourcePropType
 }
+interface BundlePair {
+  bundle1?: OCACredentialBundle
+  bundle2?: OCACredentialBundle
+}
+
 const CredentialCard: React.FC<CredentialCardProps> = ({ credential, style = {}, onPress = undefined }) => {
   const { t } = useTranslation()
   const { ColorPallet, TextTheme } = useTheme()
   const { OCABundle } = useConfiguration()
 
+  const [bundles, setBundles] = useState<BundlePair | undefined>(undefined)
+  const lang = getCurrentLanguage()
+  const metaLayer = bundles?.bundle1?.getMetaOverlay(lang) ?? bundles?.bundle2?.getMetaOverlay(lang)
+  const overlay = bundles?.bundle1?.getCardLayoutOverlay() ?? bundles?.bundle2?.getCardLayoutOverlay()
```

Comment:
The Bundles will always be undefined !?

Reply:
Yes. The bundle is resolved and loaded asynchronously when the component initializes. I am not very happy with the `bundle1`/`bundle2` pair, but this was just a quick way to get a "Default Bundle". We should probably change that later.